### ISSUE TITLE
[Bug]: Stickers from imported history don't reapply to the image properly

### ISSUE THREAD

**Issue by efalco777** [hm21/pro_image_editor#121]
### Package Version

^4.0.4

### Flutter Version

3.22.2

### Platforms

iOS

### How to reproduce?

1. Open editor
2. Add any type of sticker (like a red container from the example code) & either scale/transform it or don't touch it
3. Save & export history
4. Reopen the original image and import the history
5. Save again

**Expected:**
Upon saving the image is reapplied with the same exact sticker.

**Actual**(2 Separate results):

**1. You did not transform the sticker in any way:**
Image saves. The resulting image doesn't show the sticker. Reopening the editor with the image and newly exported history shows the sticker again.

**2. You transformed the sticker**(just a simple scale is enough):
Image editor is stuck on saving changes dialog. In console we can find an error (error log below).

### Logs (optional)

```sh
On crash:

flutter: └───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
flutter: \^[[38;5;196m┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────<…>
flutter: \^[[38;5;196m│ type 'Null' is not a subtype of type 'Uint8List' of 'value'<…>
flutter: \^[[38;5;196m├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄<…>
flutter: \^[[38;5;196m│ #0   List.add (dart:core-patch/growable_array.dart:277:14)<…>
flutter: \^[[38;5;196m│ #1   ExportStateHistory._convertLayers (package:pro_image_editor/models/import_export/export_state_history.dart:177:18)<…>
flutter: \^[[38;5;196m│ #2   <asynchronous suspension><…>
flutter: \^[[38;5;196m│ #3   ExportStateHistory.toMap (package:pro_image_editor/models/import_export/export_state_history.dart:88:7)<…>
flutter: \^[[38;5;196m│ #4   <asynchronous suspension><…>
flutter: \^[[38;5;196m│ #5   ExportStateHistory.toJson (package:pro_image_editor/models/import_export/export_state_history.dart:125:23)<…>
flutter: \^[[38;5;196m│ #6   <asynchronous suspension><…>
```

### Example code (optional)

```Dart
Reproducible with this code:
...
       stickerEditorConfigs: StickerEditorConfigs(
        enabled: true,
        buildStickers: (setLayer, _) {
          return Center(
            child: ElevatedButton(
              onPressed: () {
                final sticker = Container(
                  color: Colors.red,
                  width: 100,
                  height: 100,
                );
                setLayer(sticker);
              },
              child: Text('tap'),
            ),
          );
...
```

### Device Model (optional)

iPhone 15

**Comment by hm21** [comment]
Thank you for reporting this issue.

I tested both cases you mention, but by me everything works fine. I saw there was a resolution issue which made the quality of images lower, which I fixed, but that's all I found. Can you post for me a code example how you import the sticker with the full editor widget like `ProImageEditor.asset(...)`.

**Comment by efalco777** [comment]
Ok here's an example that I've also posted in black border issue:

```
class _FileEditor extends StatefulWidget {
  const _FileEditor({
    required this.file,
  });

  final File file;

  @override
  State<_FileEditor> createState() => _FileEditorState();
}

class _FileEditorState extends State<_FileEditor> {
  late GlobalKey<ProImageEditorState> _key;
  String? _historyJson;

  Uint8List? _editedPhotoBytes;
  bool _showEditor = true;

  @override
  void initState() {
    super.initState();
    _key = GlobalKey<ProImageEditorState>();
  }

  @override
  Widget build(BuildContext context) {
    return switch (_showEditor) {
      true => ProImageEditor.file(
          widget.file,
          key: _key,
          callbacks: ProImageEditorCallbacks(
            onImageEditingComplete: onImageEditingComplete,
          ),
          configs: ProImageEditorConfigs(
            stickerEditorConfigs: StickerEditorConfigs(
              enabled: true,
              buildStickers: (setLayer, scrollController) {
                return SizedBox(
                  height: 200,
                  width: double.infinity,
                  child: ElevatedButton(
                    child: const Text('Create sticker'),
                    onPressed: () {
                      final sticker = Container(
                        color: Colors.red,
                        width: 50,
                        height: 50,
                      );
                      setLayer(sticker);
                    },
                  ),
                );
              },
            ),
            stateHistoryConfigs: StateHistoryConfigs(
              initStateHistory: _historyJson != null ? ImportStateHistory.fromJson(_historyJson!) : null,
            ),
          ),
        ),
      false => Scaffold(
          body: Column(
            children: [
              SizedBox(
                height: 400,
                child: Image.memory(_editedPhotoBytes!),
              ),
              const Divider(),
              ElevatedButton(
                onPressed: () => _openEditor(),
                child: const Text('Reopen editor with the history'),
              )
            ],
          ),
        ),
    };
  }

  Future<void> onImageEditingComplete(bytes) async {
    final exportState = await _key.currentState?.exportStateHistory();

    _closeEditor(await exportState?.toJson(), bytes);
  }

  void _closeEditor(String? historyJson, Uint8List? bytes) {
    print('closing with ' + historyJson.toString());
    setState(() {
      _showEditor = false;
      _historyJson = historyJson;
      _editedPhotoBytes = bytes;
    });
  }

  void _openEditor() {
    print('Reopening with: ' + (_historyJson?.toString() ?? 'empty'));
    setState(() {
      _key = GlobalKey<ProImageEditorState>();
      _showEditor = true;
    });
  }
}
```

And here's video with first scenario(sticker is just applied):

https://github.com/hm21/pro_image_editor/assets/20524697/527a5a88-125d-4cf6-9d29-01751d1f7856

And here's the second(sticker is applied & transformed):

https://github.com/hm21/pro_image_editor/assets/20524697/469f4bd6-ea6e-4aa7-855b-3d7c2edf03c1

**Comment by hm21** [comment]
Thank you so much for the example, it helped me a lot to reproduce your issue. I released now version `4.0.8` which should resolve it. Please let me know when the issue persists by you after you update.
